ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -552,14 +550,16 @@ signal_list_t* netlist_expand_ast_of_module(ast_node_t** node_ref, char* instanc
         if (skip_children == false) {
             /* traverse all the children */
             for (i = 0; i < node->num_children; i++) {
-                if (child_skip_list && child_skip_list[i] == false) {
-                    sc_hierarchy* child_ref = local_ref;
-                    if (node->children[i]->types.hierarchy != NULL) {
-                        child_ref = node->children[i]->types.hierarchy;
-                    }
+                if (node->children[i]) {
```

Comment:
I agree that this should be checked, but I'm warry to add it wereas it wasnt there before and never cause any segmentation fault.
I think this will be better served by an `oassert`.
Was this added because some nodes got thrown as NULL when they werent before?

Reply:
I think this was a quick fix on my part because something NULL was being passed in, in the new commit I added an oassert and everything is fine now. My guess is I corrected what was actual wrong with another benchmark